test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,39 @@
+require "test_helper"
+
+class UsersControllerTest < ActionController::TestCase
+  include ActiveJob::TestHelper
+
+  setup do
+    login_user(users(:john))
+  end
+
+  test "GET index should be successfull" do
+    get :index
+    assert_response :success
+  end
+
+  test "POST create should create a new user" do
+    assert_difference "User.count" do
+      perform_enqueued_jobs do
+        xhr :post, :create, user: { email: "example@mail.org" }
+        delivered_email = ActionMailer::Base.deliveries.last
+        assert_includes delivered_email.to, "example@mail.org"
+      end
+    end
+
+    assert_response :success
+  end
+
+  test "GET activate should be successfull" do
+    user = users(:jane)
```

Comment:
Because you logged a user in on the `setup` block, this actions also run as user `john`. I think its a good idea to move the login_user logic away from the setup block, into the above 2 tests.